consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/ConsistencyChecker.java

```diff
@@ -202,10 +203,38 @@ public void run() {
         List<MigrationJob> jobs = new ArrayList<>();
         try {
             DataFindRequest findRequest = new DataFindRequest("migrationJob", migrationJobEntityVersion);
-            findRequest.where(and(
-                    withValue("configurationName = " + configuration.getConfigurationName()),
+            /*
+                and:
+                    whenAvailableDate <= new Date()
+                    or:
+                        jobExecutions# = 0
+                        not:
+                            jobExecutions.jobStatus $in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']
+                        and:
+                            not:
+                                jobExecutions.jobStatus $not_in [COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']
+                            jobExecutions.actualStartDate < (new Date() - expectedExecutionMilliseconds)
+
+             */
+            findRequest.where(
+                and(
                     withValue("whenAvailableDate <= " + ClientConstants.getDateFormat().format(new Date())),
-                    not(withSubfield("jobExecutions", withValue("completedFlag = true")))));
+                    or(
+                        not(
+
+                            withSubfield("jobExecutions", withValue("jobStatus $in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']"))
+                        ),
+                        and(
+                            not(
+
+                                    withSubfield("jobExecutions", withValue("jobStatus $not_in ['COMPLETED_SUCCESS', 'COMPLETED_PARTIAL']"))
+                            ),
```

Comment:
What happens for a COMPLETED_PARTIAL case? If I am reading this correctly, the migrator won't retry, so I assume it is up to the user to figure out why and schedule a retry?